src/ffi/c.rs

```diff
@@ -54,7 +56,15 @@ impl Default for StreamWrapper {
                 zalloc: Some(zalloc),
                 #[cfg(not(all(feature = "any_zlib", not(feature = "cloudflare-zlib-sys"))))]
                 zfree: Some(zfree),
-            }),
+            })),
+        }
+    }
+}
+
+impl Drop for StreamWrapper {
+    fn drop(&mut self) {
+        unsafe {
```

Comment:
Let's repeat the invariants that we expect to hold in a `SAFETY` block above `unsafe`.